staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/types_jsonschema.go

```diff
@@ -235,6 +235,16 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
+	// MessageExpression declares a CEL expression that evaluates to the validation failure message that is returned when this rule fails.
+	// Since MessageExpression is used as a failure message, it must evaluate to a string.
+	// If both Message and MessageExpression are present on a Rule, then MessageExpression will be used if validation
+	// fails. If MessageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced
+	// as if the MessageExpression field were unset.
```

Comment:
> If MessageExpression results in a runtime error, the runtime error is logged, and the validation failure message is produced as if the MessageExpression field were unset.

I would also expect that fallback if the expression evaluates to "" or null. is that what happens? if so, document and test for that

Reply:
Thanks for catching this, this wasn't tested. I'll document it and test for it as well.

Reply:
(marking as resolved, docs now note that evaluating to an empty string is not allowed, added tests as well)